### PR TITLE
EOS-14712:RAID Integrity Alerts seen recursively on all fresh deployed setups

### PR DESCRIPTION
**Pull request by shindeam**
…d setups

## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Multiple Alerts seen for RAIDIntegritySensor
  </code>
</pre>
## Problem Description
<pre>
  <code>
    RAID Integrity Alerts seen recursively on all fresh deployed setups
  </code>
</pre>
## Solution
<pre>
  <code>
    Done required changes in RAIDIntegritySensor
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes
    - [ ] No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

**Review by cdeshmukh — APPROVED** [review]

File: low-level/sensors/impl/generic/raid_integrity_data.py [hunk 1]
```diff
@@ -170,22 +173,40 @@ def _raid_health_monitor(self):
                 # Check RAID device array state is 'idle' or not
                 result = self._check_raid_state(device)
                 if result == "failed":
```
Comment (indrajitzagade) [inline low-level/sensors/impl/generic/raid_integrity_data.py:175]: Having just this check will still send alert again across system/SSPL resets. That can be handled at lower priority, I think.

File: low-level/sensors/impl/generic/raid_integrity_data.py [hunk 2]
```diff
@@ -170,22 +173,40 @@ def _raid_health_monitor(self):
...
+                    with open(fault_status_file, 'w') as fs:
+                        fs.write(data)
+                    # Retry to check RAID state
                     self._retry_execution(self._check_raid_state, device)
                 logger.debug("'idle' state is found in Raid device:{}."
                              .format(device))
 
                 # Check Mismatch count in RAID device files.
                 result = self._check_mismatch_count(device)
                 if result == "failed":
```
Comment (indrajitzagade) [inline low-level/sensors/impl/generic/raid_integrity_data.py:194]: Having just this check will still send alert again across system/SSPL resets. That can be handled at lower priority, I think.

**Review by indrajitzagade — APPROVED** [review]

**Review by indrajitzagade — CHANGES_REQUESTED** [review]
Remove state not idle related alerts

File: low-level/sensors/impl/generic/raid_integrity_data.py (hunk 2 above)
Comment (shindeam) [reply low-level/sensors/impl/generic/raid_integrity_data.py:186]: Done changes to handle such scenario

**Review by indrajitzagade — APPROVED** [review]

**Review by saurabhkhanvilkar — APPROVED** [review]

**Pull request merged**